app/src/ui/lib/sandboxed-markdown.tsx

```diff
@@ -0,0 +1,165 @@
+    const docStyle = getComputedStyle(document.body)
+    const textColor = docStyle.getPropertyValue('--text-color')
+    const backgroundColor = docStyle.getPropertyValue('--background-color')
+    const codeBackgroundColor = docStyle.getPropertyValue(
+      '--box-alt-background-color'
+    )
+    const boxBorderColor = docStyle.getPropertyValue('--box-border-color')
+
+    return `<style>
+      :root {
+        --text-color: ${textColor};
+        --background-color: ${backgroundColor};
+        --code-background-color: ${codeBackgroundColor};
+        --box-border-color: ${boxBorderColor};
+      }
+      ${css}
+    </style>`
+  }
+
+  /**
+   * We still want to be able to navigate to links provided in the markdown.
+   * However, we want to intercept them an verify they are valid links first.
+   */
+  private setupLinkInterceptor(frameRef: HTMLIFrameElement): void {
+    frameRef.addEventListener('load', () => {
+      frameRef.contentDocument?.addEventListener('click', ev => {
+        const { contentWindow } = frameRef
+
+        if (contentWindow && ev.target instanceof contentWindow.Element) {
+          const a = ev.target.closest('a')
+          if (a !== null) {
+            ev.preventDefault()
+
+            if (/^https?:/.test(a.protocol)) {
+              this.props.onMarkdownLinkClicked?.(a.href)
+            }
+          }
+        }
+      })
+    })
+  }
+
+  /**
+   * Builds a <base> tag for cases where markdown has relative links
+   */
+  private getBaseTag(baseHref: string | null): string {
+    if (baseHref == null) {
+      return ''
+    }
+
+    const base = document.createElement('base', {})
+    base.href = baseHref
+    return base.outerHTML
+  }
+
+  /**
+   * Generates an iframe with a csp allowed script of a markdown parser.
+   */
+  private mountIframeContents = async (): Promise<void> => {
+    if (this.frameRef === null) {
+      return
+    }
+
+    const styleSheet = await this.getInlineStyleSheet()
+
+    const parsedMarkdown = marked(this.props.markdown, {
+      gfm: true,
+    })
+
+    const sanitizedHTML = DOMPurify.sanitize(parsedMarkdown)
+
+    const src = `
+      <html>
+        <head>
+          ${this.getBaseTag(this.props.baseHref)}
+        </head>
+        <body>
+          ${styleSheet}
+          <div id="content">${sanitizedHTML}</div>
+        </body>
+      </html>
+    `
+
+    // We used this `Buffer.toString('base64')` approach because `btoa` could not
+    // convert non-latin strings that existed in the markedjs.
+    const b64src = Buffer.from(src, 'utf8').toString('base64')
+
+    // We are using `src` and data uri as opposed to an html string in the
+    // `srcdoc` property because the `srcdoc` property renders the html in the
+    // parent dom and therefore does not utilize the content security policy
+    // provided in the iframe's csp attribute. Additionally, we want all rendering
+    // to be isolated to our sandboxed iframe - not in the parents dom.
```

Comment:
```suggestion
    // parent dom and we want all rendering to be isolated to our sandboxed iframe.
```

Since we're not using the CSP attribute we might as well drop the specific mention of it?